internal/services/communication/helper/domain_record.go

```diff
@@ -0,0 +1,53 @@
+package helper
+
+import (
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/communication/2023-03-31/domains"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type DomainVerificationRecords struct {
+	Type  string `tfschema:"type"`
+	Name  string `tfschema:"name"`
+	Value string `tfschema:"value"`
+	TTL   int    `tfschema:"ttl"`
+}
+
+func DomainVerificationRecordsCommonSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Computed: true,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"type": {
+					Type:     pluginsdk.TypeString,
+					Computed: true,
+				},
+
+				"name": {
+					Type:     pluginsdk.TypeString,
+					Computed: true,
+				},
+
+				"value": {
+					Type:     pluginsdk.TypeString,
+					Computed: true,
+				},
+
+				"ttl": {
+					Type:     pluginsdk.TypeInt,
+					Computed: true,
+				},
+			},
+		},
+	}
+}
+
+func DomainVerificationRecordsToModel(record *domains.DnsRecord) []DomainVerificationRecords {
+	return []DomainVerificationRecords{{
+		Name:  pointer.From(record.Name),
+		Type:  pointer.From(record.Type),
+		Value: pointer.From(record.Value),
+		TTL:   int(pointer.From(record.Ttl)),
```

Comment:
As noted above, this should be an `int64` value, so we can skip the casting here.
```suggestion
		TTL:   pointer.From(record.Ttl),
```